File: src/test/java/com/github/kayjamlang/tests/expressions/value/ValueExpressionStringTest.java
package com.github.kayjamlang.tests.expressions.value;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.ValueExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class ValueExpressionStringTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("\"test\""));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(ValueExpression.class, expression.getClass());

        ValueExpression constant = (ValueExpression) expression;
        assertEquals("test", constant.value);
    }
}

File: src/test/scala/com/github/kayjamlang/tests/expressions/value/ValueExpressionStringTest.scala
package com.github.kayjamlang.tests.expressions.value

import com.github.kayjamlang.core.KayJamLexer
import com.github.kayjamlang.core.KayJamParser
import com.github.kayjamlang.core.expressions.Expression
import com.github.kayjamlang.core.expressions.ValueExpression
import org.junit.BeforeClass
import org.junit.Test
import org.junit.Assert._


object ValueExpressionStringTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("\"test\""))
  }
}

class ValueExpressionStringTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ValueExpressionStringTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[ValueExpression], expression.getClass)
    val constant = expression.asInstanceOf[ValueExpression]
    assertEquals("test", constant.value)
  }
}